Sources/IMAPlugin.swift

```diff
@@ -207,8 +207,9 @@ enum IMAState: Int, StateProtocol {
     
     public func requestAds() throws {
         guard let playerView = self.player?.view else { throw IMAPluginRequestError.missingPlayerView }
-        guard !self.config.adTagUrl.isEmpty else {
-            PKLog.debug("ad tag url is empty... can't request ads")
+        
+        if self.config.adTagUrl.isEmpty && self.config.adsResponse.isEmpty {
+            PKLog.debug("At least one should be provided: adTagUrl or adsResponse ... can't request ads")
```

Comment:
Change message to something like: "Can't request ads: at least one of adTagUrl and adsResponse must be provided".